reference/test/matrix/identity.cpp

```diff
@@ -221,4 +222,46 @@ TYPED_TEST(Identity, AppliesLinearCombinationToMixedComplex)
 }
 
 
+TYPED_TEST(Identity, ScaleDenseAddIdentity)
+{
+    using Id = typename TestFixture::Id;
+    using Vec = typename TestFixture::Vec;
+    using MixedVec = typename TestFixture::MixedVec;
+    using T = typename TestFixture::value_type;
+    auto alpha = gko::initialize<MixedVec>({2.0}, this->exec);
+    auto beta = gko::initialize<MixedVec>({-1.0}, this->exec);
+    auto identity = Id::create(this->exec, 3);
+    auto b = gko::initialize<Vec>(
+        3, {I<T>{2.0, 3.0, -1.0}, I<T>{1.0, -2.0, 2.5}, I<T>{5.0, -1.0, 4.0}},
+        this->exec);
+
+    identity->apply(alpha.get(), identity.get(), beta.get(), b.get());
+
+    GKO_ASSERT_MTX_NEAR(
+        b, l({{0.0, -3.0, 1.0}, {-1.0, 4.0, -2.5}, {-5.0, 1.0, -2.0}}), 0.0);
+}
+
+
+TYPED_TEST(Identity, ScaleCsrAddIdentity)
+{
+    using Id = typename TestFixture::Id;
+    using Vec = typename TestFixture::Vec;
+    using T = typename TestFixture::value_type;
+    using Csr = typename gko::matrix::Csr<T, int>;
+    auto alpha = gko::initialize<Vec>({2.0}, this->exec);
+    auto beta = gko::initialize<Vec>({-1.0}, this->exec);
+    auto identity = Id::create(this->exec, 3);
+    auto bd = gko::initialize<Vec>(
+        3, {I<T>{2.0, 0.0, -1.0}, I<T>{1.0, 0.0, 2.5}, I<T>{0.0, -1.0, 4.0}},
+        this->exec);
+    auto b = Csr::create(this->exec);
+    bd->convert_to(b.get());
```

Comment:
You can directly use `gko::initialize<Csr>`, which automatically converts the Dense matrix into a Csr and gives you back a Csr matrix.